serial.py

```diff
@@ -22,12 +23,17 @@ class SerialDisconnectException(SerialException):
 
 class Serial:
 
-    BAUD_MAP = { 9600: termios.B9600, 115200: termios.B115200 }
+    BAUD_MAP = {
+        9600: termios.B9600,
+        19200: 14,
+        57600: termios.B57600,
+        115200: termios.B115200
+    }
 
     def __init__(self, port, baudrate, timeout=None, **kwargs):
-        assert timeout is None, "Only no-timeout mode is supported"
         self.port = port
         self.baudrate = baudrate
+        self.timeout = None if timeout is None else timeout * 1000
```

Comment:
Did you actually test this? In MicroPython, poll() argument for "no timeout" is -1.

Reply:
Yes, I did test it.  None and -1 produce the same behavior, but we can use -1 instead.